server/catgenome/src/main/java/com/epam/catgenome/manager/vcf/reader/VcfFileReader.java

```diff
@@ -110,17 +111,25 @@ public VcfFileReader(final FileManager fileManager, final ReferenceGenomeManager
      */
     @Override
     public Track<Variation> readVariations(VcfFile vcfFile, final Track<Variation> track, Chromosome chromosome,
-                                           final Integer sampleIndex, final boolean loadInfo, final boolean collapse)
+                                           final Integer sampleIndex, final boolean loadInfo,
+                                           final boolean collapse, EhCacheBasedIndexCache indexCache)
             throws VcfReadingException {
+        double time1 = Utils.getSystemTimeMilliseconds();
         try (FeatureReader<VariantContext> reader = AbstractEnhancedFeatureReader.getFeatureReader(vcfFile.getPath(),
-                vcfFile.getIndex().getPath(), new VCFCodec(), true)) {
+                vcfFile.getIndex().getPath(), new VCFCodec(), true, indexCache)) {
             if (checkBounds(vcfFile, track, chromosome, loadInfo)) {
+                double time2 = Utils.getSystemTimeMilliseconds();
```

Comment:
Please remove this debug code from everywhere, once you've done

Reply:
Done.